bionic/deriver.py

```diff
@@ -437,33 +451,35 @@ def _attempt_to_load_task_state_results(self, task_state):
             # query.
             accessor.save_result(result)
 
-            results.append(result)
+            results_by_name[result.query.entity_name] = result
 
         if task_state.provider.attrs.should_memoize:
-            task_state.results_by_name = {
-                result.query.entity_name: result
-                for result in results
+            task_state.results_by_name = results_by_name
+            task_state.hashed_results_by_name = {
+                name: result.to_hashed_result()
+                for name, result in results_by_name.items()
             }
-        task_state.is_complete = True
 
-    def _get_results_for_complete_task_state(self, task_state):
+        return results_by_name
+
+    def _get_hashed_results_for_complete_task_state(self, task_state):
```

Comment:
Is there any special reason `hashed_results_by_name` needs to be set lazily?

Reply:
Now that I think about it, we don't need to set it lazily.